### PR TITLE
save-demo-agent template

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR is a part of #1791

### What's done:
 * Added new module: `save-demo-agent`
 * Added `ktor` server to `libs.versions.toml`
 * Implemented simple server with `ktor` in `save-demo-agent`
 * Added test on server startup

### CODE REVIEW

File: save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import io.ktor.server.application.*
+import io.ktor.server.cio.*
+import io.ktor.server.engine.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import okio.FileSystem
+
+@Suppress("MagicNumber")
+fun main() {
+    runServerOnPort(23456).start()
```
Comment (kgevorkyan) [inline save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt:18]: make port configurable via properties file?

File: save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+/**
+ * @param port port number
+ * @return [CIOApplicationEngine]
+ */
+@Suppress("MagicNumber")
+internal fun runServerOnPort(port: Int) = ProcessBuilder(true, FileSystem.SYSTEM).let { pb ->
```
Comment (kgevorkyan) [inline save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt:26]: why do we need processbuilder for running server? we can't do it directly via ktor?

Comment (sanyavertolet) [reply save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt:26]: We use process builder only to run shell command for now.

File: save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/Main.kt:18]: Will be done in next PR

**Review by kgevorkyan — APPROVED** [review]

File: save-demo-agent/src/nativeTest/kotlin/com/saveoourtool/save/demo/agent/MainTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlin.test.*
+
+class MainTest {
+    private val serverUrl = "localhost"
+    private val serverPort = 23456
+    private val server = runServerOnPort(23456)
```
Comment (kgevorkyan) [inline save-demo-agent/src/nativeTest/kotlin/com/saveoourtool/save/demo/agent/MainTest.kt:18]:
```kotlin
private val server = runServerOnPort(serverPort)
```

File: save-demo-agent/src/nativeTest/kotlin/com/saveoourtool/save/demo/agent/MainTest.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlin.test.*
+
+class MainTest {
```
Comment (kgevorkyan) [inline save-demo-agent/src/nativeTest/kotlin/com/saveoourtool/save/demo/agent/MainTest.kt:15]: guess. it should be renamed in future

Comment (sanyavertolet) [reply save-demo-agent/src/nativeTest/kotlin/com/saveoourtool/save/demo/agent/MainTest.kt:15]: Surely

**Pull request merged**